src/librbd/AioImageRequestWQ.cc

```diff
@@ -111,6 +111,10 @@ void AioImageRequestWQ::aio_read(AioCompletion *c, uint64_t off, uint64_t len,
 
   RWLock::RLocker owner_locker(m_image_ctx.owner_lock);
 
+  if (m_image_ctx.clone_copy_on_read && is_lock_required()) {
```

Comment:
Probably better to combine this with the existing logic in ```librbd::image::RefreshRequest``` to avoid having this flag cleared should journaling be disabled while a read IO is queued. Probably can avoid the other changes in this file if that change is made.